src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/CustomiseHearingBundleHandler.java

```diff
@@ -0,0 +1,281 @@
+
+
+@Component
+public class CustomiseHearingBundleHandler  implements PreSubmitCallbackHandler<AsylumCase> {
+    private final BundleRequestExecutor bundleRequestExecutor;
+    private final Appender<DocumentWithMetadata> appender;
+    private final DateProvider dateProvider;
+    private final String emBundlerUrl;
+    private final String emBundlerStitchUri;
+
+    public CustomiseHearingBundleHandler(
+            @Value("${emBundler.url}") String emBundlerUrl,
+            @Value("${emBundler.stitch.uri}") String emBundlerStitchUri,
+            BundleRequestExecutor bundleRequestExecutor,
+            Appender<DocumentWithMetadata> appender,
+            DateProvider dateProvider
+    ) {
+        this.emBundlerUrl = emBundlerUrl;
+        this.emBundlerStitchUri = emBundlerStitchUri;
+        this.bundleRequestExecutor = bundleRequestExecutor;
+        this.appender = appender;
+        this.dateProvider = dateProvider;
+
+    }
+
+    public boolean canHandle(
+            PreSubmitCallbackStage callbackStage,
+            Callback<AsylumCase> callback
+    ) {
+        requireNonNull(callbackStage, "callbackStage must not be null");
+        requireNonNull(callback, "callback must not be null");
+
+        return callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
+                && callback.getEvent() == Event.CUSTOMISE_HEARING_BUNDLE;
+    }
+
+    public PreSubmitCallbackResponse<AsylumCase> handle(
+            PreSubmitCallbackStage callbackStage,
+            Callback<AsylumCase> callback
+    ) {
+        if (!canHandle(callbackStage, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
+        }
+
+        final AsylumCase asylumCase =
+                callback
+                        .getCaseDetails()
+                        .getCaseData();
+
+        prepareDocuments(asylumCase, CUSTOM_HEARING_DOCUMENTS,HEARING_DOCUMENTS);
+        prepareDocuments(asylumCase, CUSTOM_LEGAL_REP_DOCUMENTS,LEGAL_REPRESENTATIVE_DOCUMENTS);
+        prepareDocuments(asylumCase, CUSTOM_ADDITIONAL_EVIDENCE_DOCUMENTS,ADDITIONAL_EVIDENCE_DOCUMENTS);
+        prepareDocuments(asylumCase, CUSTOM_RESPONDENT_DOCUMENTS,RESPONDENT_DOCUMENTS);
+
+        asylumCase.clear(AsylumCaseFieldDefinition.HMCTS);
+        asylumCase.write(AsylumCaseFieldDefinition.HMCTS,"[userImage:hmcts.png]");
+
+        asylumCase.clear(AsylumCaseFieldDefinition.CASE_BUNDLES);
+        asylumCase.write(AsylumCaseFieldDefinition.BUNDLE_CONFIGURATION, "iac-hearing-bundle-config.yaml");
+        asylumCase.write(AsylumCaseFieldDefinition.BUNDLE_FILE_NAME_PREFIX, getBundlePrefix(asylumCase));
+
+        final PreSubmitCallbackResponse<AsylumCase> response = bundleRequestExecutor.post(callback, emBundlerUrl + emBundlerStitchUri);
+
+        final AsylumCase responseData = response.getData();
+
+        //Customise the folders before returning
+        CaseDetails<AsylumCase> caseDetailsBefore = callback.getCaseDetailsBefore().orElse(null);
```

Comment:
We can't use caseDetailsBefore from callback. It is not the version one event before our event. It is like reference to previous, but we need to have reference to current.